tests/operators/test_python.py

```diff
@@ -817,6 +822,31 @@ def f():
 
         self._run_as_operator(f, requirements=['funcsigs', 'dill'], system_site_packages=False)
 
+    def test_private_repo_requirements(self):
+
+        with create_session() as session:
+            self._clean_stop_pypi_server()
+            self._initiate_pypi_server()
+            session.execute("delete from connection where conn_id = 'private_repo_test_conn';")
+            cn = Connection("private_repo_test_conn",
```

Comment:
There is no need to create the connection. It's enough to patch the method with `AIRFLOW__CONN__NN` environment variables (https://airflow.apache.org/docs/apache-airflow/stable/howto/connection.html#storing-a-connection-in-environment-variables  - also there is an example of how you can generate the connection url from the raw values - incluidng percent-encoding etc. https://airflow.apache.org/docs/apache-airflow/stable/howto/connection.html#generating-a-connection-uri) . Another problem with this implementation is that it creates side-effect - the connection remains in the database after the test is run. If you termporarily patch env variable for the test, it will work faster and will not leave a side effect.

Reply:
See also docs about mocking connection: https://airflow.apache.org/docs/apache-airflow/stable/best-practices.html#mocking-variables-and-connections